### PR TITLE
UI animations

### PR DESCRIPTION
**Pull request by serafy0**
### What this PR does (required):
- added grow animations to some components

### Screenshots / Videos (required):
https://user-images.githubusercontent.com/29197126/132955603-26221031-bd22-4680-b0b2-60b0d00a54ff.mov

### Any issues with the current functionality (optional):
- this is still a work on progress.

### CODE REVIEW

**Comment by serafy0** [comment]
added skeleton loading animation for notifications 

https://user-images.githubusercontent.com/29197126/133086160-be983fe3-609d-4543-9d83-6510a9d738e0.mov


https://user-images.githubusercontent.com/29197126/133085635-eacc9364-a6e5-4162-832d-8819025d2132.mov

File: client/src/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+              <NotificationsMenu />
+              <Box padding={1} marginLeft={1}>
+                <Button>My jobs</Button>
+              </Box>
+              <Box padding={1} marginLeft={1}>
+                <Button>Messages</Button>
+              </Box>
+
+              {/*{if profile.type == sitter}*/}
```
Comment (moffatethan) [inline client/src/components/NavBar/NavBar.tsx:32]: Remove please 😄

File: client/src/components/NavBar/NotificationsMenu/NotificationsMenuItems/NotificationsMenuItems.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export default function NotificationsMenuItems({ notifications }: Props): JSX.Element {
+  const classes = useStyles();
+  const defaultLogoURL = 'https://res.cloudinary.com/dalisapxa/image/upload/v1630880229/DEV/logo_nywmrf.png';
+
+  return (
+    <Box>
+      <Paper>
+        {notifications &&
+          notifications.map((notification, index) => (
+            <Box key={index} paddingLeft={'4px'}>
```
Comment (moffatethan) [inline client/src/components/NavBar/NotificationsMenu/NotificationsMenuItems/NotificationsMenuItems.tsx:18]: Given we can receive notifications and so indexes change this is actually an anti-pattern to use indexes as keys - please use something more unique such as the notification id.

File: client/src/helpers/APICalls/Notification.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const postFetchOptions: FetchOptions = {
+  method: 'POST',
+  credentials: 'include',
+};
+
+const getAllNotifications = async (): Promise<[Notification]> => {
+  return await fetch(`/notification/all`, fetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({ error: { message: 'Unable to connect to server. Please try again' } }));
+};
```
Comment (moffatethan) [inline client/src/helpers/APICalls/Notification.ts:17]: I would just use async/await here as opposed to mixing .then and async/await.

File: client/src/helpers/APICalls/Notification.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+const getAllUnreadNotifications = async (): Promise<[Notification]> => {
+  return await fetch(`/notification/all-unread`, fetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({ error: { message: 'Unable to connect to server. Please try again' } }));
+};
```
Comment (moffatethan) [inline client/src/helpers/APICalls/Notification.ts:23]: Same here

File: client/src/helpers/APICalls/Notification.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+const markAsRead = async (id: string) => {
+  return await fetch(`/notification/seen/${id}`, postFetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({ error: { message: 'Unable to connect to server. Please try again' } }));
+};
```
Comment (moffatethan) [inline client/src/helpers/APICalls/Notification.ts:29]: Same here

File: client/src/pages/Profile/Profile.tsx [hunk 6]
```diff
@@ -1,15 +1,91 @@
...
+import getProfile from '../../helpers/APICalls/getProfile';
+import { ProfileApiData } from '../../interface/ProfileApiData';
+import { FormikHelpers } from 'formik';
+import updateProfile from '../../helpers/APICalls/updateProfile';
 
 export default function Profile(): JSX.Element {
   const { loggedInUser } = useAuth();
+  const { path, url } = useRouteMatch();
+
+  console.log('path:', path, 'url:', url);
```
Comment (moffatethan) [inline client/src/pages/Profile/Profile.tsx:19]: Remove please, is this duplicated from #67

**Review by moffatethan — COMMENTED** [review]
Please address my comments and going forward please be cautious of the following.

1. We should be naming our PR titles more descriptively - "UI animations" but it has notifications work in it. The context is hard and makes it hard for me to know what I am reviewing exactly.
2. Ensuring we aren't pushing duplicated code for review (#67) code is in this PR.

**Comment by serafy0** [comment]
If i changed the base branch to `main` again will most of those duplicated code be removed ?
I think the problem i did was branching from https://github.com/hatchways/team-naan-bread/pull/67 and then merging https://github.com/hatchways/team-naan-bread/pull/63 into this pull request

**Review by moffatethan — CHANGES_REQUESTED** [review]
So close, please work on the merge conflicts and then I will give approval.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**